clients/src/main/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImpl.java

```diff
@@ -587,6 +599,18 @@ public boolean shouldSkipHeartbeat() {
         return state == MemberState.UNSUBSCRIBED || state == MemberState.FATAL;
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void transitionToStaled() {
+        transitionTo(MemberState.PREPARE_LEAVING);
```

Comment:
Looks to me this is not really needed here? We use this state when we are taking some action before sending the last HB (mainly callbacks), but in this case there is no async operation to do, just send the last HB right?